### PR TITLE
Update samtools view

### PR DESCRIPTION
**Pull request by priyanka-surana**
- Adds functionality to convert from bam to cram and vice versa
- Adds test to check cram to bam
- Removes csi output

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [ x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ x] Emit the `versions.yml` file.
- [ x] Follow the naming conventions.
- [ x] Follow the parameters requirements.
- [ x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/samtools/view/meta.yml [hunk 1]
```diff
@@ -39,23 +39,16 @@ output:
...
       type: file
-      description: filtered/converted BAM/SAM file
-      pattern: "*.{bam,sam}"
-  - cram:
-      type: file
-      description: filtered/converted CRAM file
-      pattern: "*.cram"
+      description: filtered/converted BAM/SAM/CRAM file
+      pattern: "*.{bam,sam,cram}"
   - versions:
```
Comment (jfy133) [inline modules/samtools/view/meta.yml:46]: These are three separate output channels according to the module `main.nf`

Comment (priyanka-surana) [reply modules/samtools/view/meta.yml]: updated

**Review by jfy133 — APPROVED** [review]
LGTM for when tests pass :)

**Comment by priyanka-surana** [comment]
The singularity and conda output file md5sums are different for the `convert` test. But I am not sure why, since the headers are identical.

**Pull request merged**